tools/python/booking_hotels.py

```diff
@@ -0,0 +1,196 @@
+        self.login = login
+        self.password = password
+        self.baseConfig = {
+            "headers": {
+                "Content-Type": "application/json",
+                "Authorization": "Basic " + base64.encodestring(
+                    "{login}:{password}".format(login=self.login, password=self.password)).replace('\n', '')
+            },
+            "url": 'https://distribution-xml.booking.com/json/bookings'}
+        self.checkMinute = 0
+        self.requestPerMinute = 0
+        self.requestLimit = 15  # request per minute
+
+    def call(self, function, params=None):
+        self.requestPerMinute += 1
+        now = datetime.utcnow()
+
+        if self.requestPerMinute >= self.requestLimit:
+            waittime = 60 - now.second
+            logging.warning("Limit for request per minute exceeded. Wait for: {0} sec.".format(waittime))
+            time.sleep(waittime)
+            now = datetime.utcnow()
+
+        if self.checkMinute != now.minute:
+            self.requestPerMinute = 0
+            self.checkMinute = now.minute
+
+        payload = ''
+        try:
+            p = "" if not params else '?' + "&".join(
+                ["{key}={value}".format(key=k, value=v) for (k, v) in params.iteritems()])
+            url = "{base}.{func}{params}".format(base=self.baseConfig["url"], func=function, params=p)
+            logging.debug("{0} {1} API call:{2}".format(self.checkMinute, self.requestPerMinute, url))
+            request = urllib2.Request(url, None, self.baseConfig["headers"])
+            stream = urllib2.urlopen(request)
+            payload = stream.read()
+            return json.loads(payload)
+
+        except Exception as e:
+            logging.error('Error: {0} Context: {1}'.format(e, payload))
+            return None
+
+
+def make_record(src, rate):
+    return Hotel(
+        int(src['hotel_id']),
+        float(src['location']['latitude']),
+        float(src['location']['longitude']),
+        src['name'],
+        src['address'],
+        int(src['class']),
+        rate,
+        src['ranking'],
+        src['review_score'],
+        src['url']
+    )
+
+
+def download(user, password, path):
+    api = BookingApi(user, password)
+
+    maxrows = 1000
+    countries = api.call("getCountries", dict(languagecodes='en'))
+    for country in countries:
+        countrycode = country['countrycode']
+        logging.info(u'{0} {1}'.format(countrycode, country['name']))
+
+        counter = 0
+        allhotels = []
+        while True:
+            hotels = api.call('getHotels',
+                              dict(new_hotel_type=1, offset=counter, rows=maxrows, countrycodes=countrycode))
+            if isinstance(hotels, dict) and 'ruid' in hotels:
+                logging.error('{0} Code: {1}'.format(hotels['message'], hotels['code']))
+                exit(1)
+
+            for hotel in hotels:
+                allhotels.append(hotel)
+
+            counter += len(hotels)
```

Comment:
What's the purpose to keep counter variable? Why not to directly use len(allhotels)?